internals/overlord/overlord.go

```diff
@@ -321,6 +328,25 @@ func (o *Overlord) ensureTimerSetup() {
 	o.pruneTicker = time.NewTicker(pruneInterval)
 }
 
+func (o *Overlord) ensureTimerStopClearChannel() {
```

Comment:
This is an awkward name. How about just `stopEnsureTimer`?

Reply:
The naming scheme in overlord is this:

```
ensureTimer
ensureTimerSetup
ensureTimerReset
```

So I think that simply making this:

```ensureTimerStop```

is perhaps a good enough improvement ? Let me know what you prefer. This is now also closer aligned with the fact that from Go 1.23 this method can be replaced by simply timer.Stop, but for backwards compatibility of earlier Go versions, we need to explicitly clear the channel right now.

Reply:
@benhoyt Let me know if my argument is acceptable for you.

Reply:
Following a much cleaner proposal from @pedronis, this function is no longer needed. Resolving.